google-cloud-pub-sub/src/main/scala/akka/stream/alpakka/googlecloud/pubsub/Model.scala

```diff
@@ -4,10 +4,23 @@
 
 package akka.stream.alpakka.googlecloud.pubsub
 
+import java.time.Instant
+
 import scala.collection.immutable
 import scala.collection.JavaConverters._
 
-final case class PubSubMessage(data: String, messageId: String)
+final case class PubSubMessage(data: String,
+                               messageId: String,
+                               attributes: Option[immutable.Map[String, String]] = None,
+                               publishTime: Option[Instant] = None) {
+
+  def withAttributes(attributes: java.util.Map[String, String]): PubSubMessage =
+    copy(attributes = Some(attributes.asScala.toMap))
+
+  def withPublishTime(publishTime: Instant): PubSubMessage =
+    copy(publishTime = Some(publishTime))
+
+}
```

Comment:
We'd need a companion object with a Java-friendly `create` method to get by without Scala Option in the interface:
`def create(data: String, messageId: String): PubSubMessage`